### ISSUE TITLE
Nonexistent steps are flagged as valid if they contain the full text of a real step

### ISSUE THREAD

**Issue by mtkennerly** [alexkrechik/VSCucumberAutoComplete#222]
**Describe the bug**
A nonexistent step "Given x and y" will be satisfied by a definition for "Given x" or a definition for "Given and y".

This is related to #223, but slightly different. The plugin assumes regex step definitions, but Behave's default style is non-regex. Even if you switch Behave to regex mode, it forbids you from using `^` and `$` because it inserts them automatically, and since the plugin doesn't see `^` and `$`, it assumes that partial matches are okay. The hypothetical `cucumberautocomplete.stepDefinitionsUseRegex` option I mentioned in #223 would solve this when in non-regex mode - but to solve missing `^` and `$` in regex mode, you would need an option like `cucumberautocomplete.stepDefinitionsRegexAlwaysWhole` (`false` for current behavior).

**To Reproduce**
See Gherkin/steps below.

**Expected behavior**
A step should only be found if all of its phrasing is present in the definition.

**Screenshots**
![image](https://user-images.githubusercontent.com/4934192/48990693-7aa7a780-f0fd-11e8-89ca-2a7f0fe57be6.png)

**Cucumberautocomplete part of VSCode settings:**
```
{
    "cucumberautocomplete.steps": ["steps/*.py"],
    "cucumberautocomplete.strictGherkinCompletion": false
}
```
The same thing also happens with `cucumberautocomplete.strictGherkinCompletion` set to `true`.

**Step definition:**
```python
from behave import *

@given("some multiword precondition")
def step_impl(context):
    pass
```

**Gherkin step line**
If applicable, add step line from gherkin file
```gherkin
Feature: Foo
  Scenario: A
    Given some multiword precondition whoops this isn't defined  # <-- jumps to definition for "Given some multiword precondition"
```

**Comment by mtkennerly** [comment]
On second thought, this is related to #223, since the plugin is assuming regex definitions, but Behave doesn't use regex by default. In fact, even if you switch Behave to regex mode, it forbids you from using `^` or `&` because it inserts them automatically. And since the plugin doesn't see `^` or `&`, it infers that partial matches are okay. I'll close this ticket in favor of #223 and expand that one's description.

**Comment by alexkrechik** [comment]
Looks like, that it would be better to fix such things by providing some language/mode option.

**Comment by kapoorlakshya** [comment]
I am experiencing the same when using Cucumber in Ruby. Also, the steps are not matching the correct step definition. For example:

```
And I set the 'Name' text field as 'Lakshya'
```

takes me to this line within a different step:

```
step "I set the '#{hash[:field]}' #{field_type} as '#{hash[:value]}'"
```

instead of:

```
When /^I (?:set|update) the '(.+)' ((?:date|time|text) field) (?:as|to) '(.+)'(?: and press the '(.+)' key)?$/ do |field, field_type, val, action|
```

**Comment by alexkrechik** [comment]
Should be fixed with `"cucumberautocomplete.pureTextSteps": true` option using after 2.16.0 release.

**Comment by alexkrechik** [comment]
This issue is to be fixed or implemented in the 3.0.0 version of the extension. If this issue is unresolved or there are additional notes, please create a new ticket with the necessary details.
